### PR TITLE
Enabling on-chain pricing by US Dollars

### PR DESCRIPTION
**Pull request by tjayrush**

### CODE REVIEW

File: src/ui/views/Dashboard/Dashboard.tsx [hunk 1]
```diff
@@ -138,17 +137,16 @@ export const DashboardView = () => {
           }),
           loading: false,
         });
+        setLoading(false);
```
Comment (dszlachta) [inline src/ui/views/Dashboard/Dashboard.tsx:140]: I wonder if we should take all requests into consideration when showing loading, e.g. `someRequestsLoading = listLoading || exportLoading`

**Review by dszlachta — APPROVED** [review]
Looks good 👍

File: src/ui/views/Dashboard/Dashboard.tsx (hunk 1 above)
Comment (tjayrush) [reply src/ui/views/Dashboard/Dashboard.tsx:140]: I want loading to show only for slow queries. The query for record count is very fast, so showing it each time I switch subviews (either the left side or right side tabs), it flashed to loading, but then returned immeadiatly when it found that nRecords didn't change. I also only want to show loading until a few records show up on the history display (or charts). That's why the `< 10` line.

**Comment by dszlachta** [comment]
Ah, I get it. This will do for now, but I think that we may want to create debounce loader component, that would only show loading if it’s taking some time. It may be hard to predict which requests can be slow

Dawid Szlachta

**Comment by tjayrush** [comment]
Yes. I'm sure there's a better way. The goal is "only show loading if it's taking XXXX amount of time", where 'XXX == annoying' or 'XXX == user think the app is slow'

We always want the user to think the app is fast even though in many cases the back end is slow -- because of queries to the node.

**Pull request merged**